distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -1181,7 +1208,44 @@ public void testRetrievalReliablility() throws Exception {
                         Condition.parameter("request", "GetRecordById"),
                         Condition.parameter("id", metacardId));
 
-        // Add CometD notification and activity assertions when DDF-2272 ihas been addressed.
+        expect("Waiting for notifications").within(10, SECONDS)
+                .until(() -> localhostCometDClient.getMessages(NOTIFICATIONS_CHANNEL)
+                        .size() == 3);
+        expect("Waiting for activities").within(10, SECONDS)
+                .until(() -> localhostCometDClient.getMessages(ACTIVITIES_CHANNEL)
+                        .size() == 9);
+
+        List<String> notifications = localhostCometDClient.getMessagesInAscOrder(NOTIFICATIONS_CHANNEL);
+        assertThat(notifications.size(), is(3));
+        CometDMessageValidator.verifyNotification(JsonPath.from(notifications.get(0)), filename,
+                "Resource retrieval retrying after 1 bytes. Attempt 1 of 3.", "retrying");
+        CometDMessageValidator.verifyNotification(JsonPath.from(notifications.get(1)), filename,
+                "Resource retrieval retrying after 1 bytes. Attempt 2 of 3.", "retrying");
+        CometDMessageValidator.verifyNotification(JsonPath.from(notifications.get(2)), filename,
+                String.format("Resource retrieval completed, %d bytes retrieved. ",
+                        resourceData.length()), "complete");
+
+        List<String> activities = localhostCometDClient.getMessagesInAscOrder(ACTIVITIES_CHANNEL);
+        assertThat(activities.size(), is(9));
+        CometDMessageValidator.verifyActivity(JsonPath.from(activities.get(0)), filename,
```

Comment:
unfortunately we are testing to make sure we get the correct cometd messages, so we have to look at the message.